indico/modules/events/papers/controllers/common.py

```diff
@@ -0,0 +1,41 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2017 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import os
+
+from indico.modules.events.util import ZipGeneratorMixin
+from indico.util.fs import secure_filename
+
+
+class PapersDownloadAttachmentsMixin(ZipGeneratorMixin):
```

Comment:
Will this be used in more than one place? If not let's move the logic to the RH itself instead of a separate mixin

If we keep the mixin, I'd go for nicer name.. e.g. `DownloadPapersMixin`